tensorflow_datasets/core/download/kaggle.py

```diff
@@ -144,10 +157,175 @@ def _run_kaggle_command(command_args, competition_name):
     if output.startswith(b"404"):
       logging.error(_NOT_FOUND_ERR_MSG, competition_name)
       raise
+    if output.startswith(b'403'):
+      logging.error(_NOT_ACCEPT_RULES_ERR_MSG, competition_name)
+      raise
     logging.error(_ERR_MSG, competition_name)
     raise
 
 
 def _log_command_output(output, error=False):
   log = logging.error if error else logging.info
   log("kaggle command output:\n%s", tf.compat.as_text(output))
+
+
+class KaggleSearch(object):
+  """Searcher for a Kaggle Competitions and Datasets."""
+  def __init__(self, where='competitions', search=None, group=None,
+               category=None,
+               sort_by_competitions=None, sort_by_datasets=None, size=None,
+               file_type=None, tag=None, user=None):
+    self.search = search
+    self.group = group
+    self.category = category
+    self.sort_by_competitions = sort_by_competitions
+    self.where = where
+    self.sort_by_datasets = sort_by_datasets
+    self.size = size
+    self.file_type = file_type
+    self.tag = tag
+    self.user = user
+
+  # TODO add page number
+  @property
+  def searching(self):
+    """Run kaggle search with subprocess."""
+    command = ["kaggle", self.where, 'list']
+
+    competitions_parameters = {'--search': self.search, '--group': self.group,
+                               '--category': self.category,
+                               '--sort-by': self.sort_by_competitions}
+    datasets_parameters = {'--search': self.search,
+                           '--sort-by': self.sort_by_datasets,
+                           '--size': self.size, '--file-type': self.file_type,
+                           '--tags': self.tag, '--user': self.user}
+
+    if self.where == 'competitions':
+      parameters = competitions_parameters
+    else:
+      parameters = datasets_parameters
+
+    for i, j in parameters.items():
+      if j is not None or '':
+        command.extend([i, j])
+
+    if self.where != 'competitions':
+      # Fit to the dataset table.
+      display.display(display.HTML("<style>.container { width:80% !important; }"
+                                   "</style>"))
+
+    output = _run_kaggle_command(command)
+    return output
+
+
+def search_and_download():
```

Comment:
I would put "interactive" or "colab" or "notebook" in the name to make it more clear that it is intended to use in a notebook.